src/machine-c-api.h

```diff
@@ -82,1894 +83,528 @@ typedef enum { // NOLINT(modernize-use-using)
-
-/// \brief Semantic version
-typedef struct { // NOLINT(modernize-use-using)
-    uint32_t major;
-    uint32_t minor;
-    uint32_t patch;
-    const char *pre_release;
-    const char *build;
-} cm_semantic_version;
-
-/// \brief Memory range description
-typedef struct { // NOLINT(modernize-use-using)
-    uint64_t start;
-    uint64_t length;
-    const char *description;
-} cm_memory_range_descr;
-
-/// \brief Memory range description array
-typedef struct { // NOLINT(modernize-use-using)
-    cm_memory_range_descr *entry;
-    size_t count;
-} cm_memory_range_descr_array;
-
-// ---------------------------------
+/// \brief Access log types.
+typedef enum {
+    CM_ACCESS_LOG_TYPE_PROOFS = 1 << 1,      ///< Includes proofs
+    CM_ACCESS_LOG_TYPE_ANNOTATIONS = 1 << 2, ///< Includes annotations
+    CM_ACCESS_LOG_TYPE_LARGE_DATA = 1 << 3   ///< Includes data bigger than 8 bytes
+} CM_ACCESS_LOG_TYPE;
+
+/// \brief List of CSRs to use with cm_read_csr and cm_write_csr.
+typedef enum CM_CSR {
+    CM_CSR_PC,
+    CM_CSR_FCSR,
+    CM_CSR_MVENDORID,
+    CM_CSR_MARCHID,
+    CM_CSR_MIMPID,
+    CM_CSR_MCYCLE,
+    CM_CSR_ICYCLEINSTRET,
+    CM_CSR_MSTATUS,
+    CM_CSR_MTVEC,
+    CM_CSR_MSCRATCH,
+    CM_CSR_MEPC,
+    CM_CSR_MCAUSE,
+    CM_CSR_MTVAL,
+    CM_CSR_MISA,
+    CM_CSR_MIE,
+    CM_CSR_MIP,
+    CM_CSR_MEDELEG,
+    CM_CSR_MIDELEG,
+    CM_CSR_MCOUNTEREN,
+    CM_CSR_MENVCFG,
+    CM_CSR_STVEC,
+    CM_CSR_SSCRATCH,
+    CM_CSR_SEPC,
+    CM_CSR_SCAUSE,
+    CM_CSR_STVAL,
+    CM_CSR_SATP,
+    CM_CSR_SCOUNTEREN,
+    CM_CSR_SENVCFG,
+    CM_CSR_ILRSC,
+    CM_CSR_IFLAGS,
+    CM_CSR_IUNREP,
+    CM_CSR_CLINT_MTIMECMP,
+    CM_CSR_PLIC_GIRQPEND,
+    CM_CSR_PLIC_GIRQSRVD,
+    CM_CSR_HTIF_TOHOST,
+    CM_CSR_HTIF_FROMHOST,
+    CM_CSR_HTIF_IHALT,
+    CM_CSR_HTIF_ICONSOLE,
+    CM_CSR_HTIF_IYIELD,
+    CM_CSR_UARCH_PC,
+    CM_CSR_UARCH_CYCLE,
+    CM_CSR_UARCH_HALT_FLAG,
+    CM_CSR_COUNT,              ///< Number of CSRs
+    CM_CSR_UNKNOWN,            ///< Unknown CSR
+    CM_CSR_HTIF_TOHOST_DEV,    ///< View of DEV field from HTIF_TOHOST
+    CM_CSR_HTIF_TOHOST_CMD,    ///< View of CMD field from HTIF_TOHOST
+    CM_CSR_HTIF_TOHOST_DATA,   ///< View of DATA field from HTIF_TOHOST
+    CM_CSR_HTIF_FROMHOST_DEV,  ///< View of DEV field from HTIF_FROMHOST
+    CM_CSR_HTIF_FROMHOST_CMD,  ///< View of CMD field from HTIF_FROMHOST
+    CM_CSR_HTIF_FROMHOST_DATA, ///< View of DATA field from HTIF_FROMHOST
+    CM_CSR_IFLAGS_PRV,         ///< View of PRV field from IFLAGS
+    CM_CSR_IFLAGS_X,           ///< View of X field from IFLAGS
+    CM_CSR_IFLAGS_Y,           ///< View of Y field from IFLAGS
+    CM_CSR_IFLAGS_H,           ///< View of H field from IFLAGS
+} CM_CSR;
+
+/// \brief Machine instance handle.
+/// \details cm_machine* is handle used to pass the machine object when calling C API functions.
+typedef struct cm_machine cm_machine;
+
+/// \brief Machine hash array.
+typedef uint8_t cm_hash[CM_MACHINE_HASH_BYTE_SIZE];
```

Comment:
This can be error prone when used as a function argument, for instance.

Prefer something like this.
```
typedef struct {
    uint8_t data[CM_MACHINE_HASH_BYTE_SIZE];
} cm_hash;
```